addons/udes_sale_stock/models/sale_order.py

```diff
@@ -393,6 +393,53 @@ def extract_exception_data(err):
 
         return True
 
+    def merge_confirm_orders_reserve_stock(
+        self,
+        confirm_orders_batch_size=1000,
+        confirm_orders_commit_size=500,
+        finish=False,
+    ):
+        """Confirming orders and Reserving Stock in a single cron job so the processes can be
+        run simultaneously.
+        The process will be like below:
+        1. Reserving stock if there are not orders to confirm.
+        2. Splitting all orders to confirm in sizes with confirm_orders_batch_size.
+        3. For every orders_to_confirm batch ,confirming
+        and committing orders with in confirm_orders_commit_size.
+        4. After confirming orders_to_confirm batch reserving stock.
+        5. If needed to confirm only 1 confirm_orders_batch_size finish variable
+        will get out and finish the cron job.
+        6. Not confirmed orders and not reserved stock will be picked in the
+        next cron job if finish is set to true.
+        :param int confirm_orders_batch_size:
+            Number of orders will be confirmed before reserving stock.
+        :param int confirm_orders_commit_size:
+            Number of orders will be confirmed before committing.
+        :param boolean finish:
+            Flag if enabled will finish the cron job after the first iteration and pick the
+            remaining orders to confirm on next cron run.
+        """
+        SaleOrder = self.env["sale.order"]
+        StockPicking = self.env["stock.picking"]
+
+        orders_to_confirm = SaleOrder.get_orders_to_confirm()
+        # If there aren't any orders to confirm just run the reserve stock cron
+        if not orders_to_confirm:
+            StockPicking.reserve_stock()
+        for _, batched_orders in orders_to_confirm.batched(size=confirm_orders_batch_size):
```

Comment:
nit: I know _ is not used in the for loop but maybe you could give it a different name like batch_range, just so it doesn't get confused with the translator function _()

Reply:
When it is not used if have a variable name, it would complain that unused variable name. In fact we dont need it to use. Seen in other cases the same approach thats why did in the same way

Reply:
When it is not used if have a variable name, it would complain that unused variable name. In fact we dont need it to use. Seen in other cases the same approach thats why did in the same way